### PR TITLE
[CARBONDATA-3808] Added documentation for cdc and scd scenarios

### PR DESCRIPTION
**Pull request by Indhumathi27**
### Why is this PR needed?
 Added documentation for cdc and scd scenarios
 
 ### What changes were proposed in this PR?
Added documentation for cdc and scd scenarios
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2971/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1253/

File: docs/scd-and-cdc-guide.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          .whenNotMatched(<condition>)
+          .insertExpr(insertMap)
+          .whenNotMatchedAndExistsOnlyOnTarget(<condition>)
+          .delete()
+          .insertHistoryTableExpr(insertMap_d, <table_name>)
+          .execute()
+```
+**NOTE:** SQL syntax for merge is not yet supported.
+
+##### Example code sample to implement ccd scenario
```
Comment (ajantha-bhat) [inline docs/scd-and-cdc-guide.md:48]: Just give a link to `MergeTestCase.scala` and test case name
No need have same thing here

File: docs/scd-and-cdc-guide.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+than changing regularly on a time basis.
+
+SCD and CDC data changes can be merged to a carbon dataset online using the data frame level `MERGE` API.
+
+#### MERGE API
+
+Below API merges the datasets online and applies the actions as per the conditions. 
+
+```
+  targetDS.merge(sourceDS, <condition>)
```
Comment (ajantha-bhat) [inline docs/scd-and-cdc-guide.md:35]: Need to explain what each and every API does here and when to use it. Because these are added newly by carbon

Comment (Indhumathi27) [reply docs/scd-and-cdc-guide.md:35]: UPdated the PR

File: docs/scd-and-cdc-guide.md (hunk 1 above)
Comment (Indhumathi27) [reply docs/scd-and-cdc-guide.md]: done

**Comment by ajantha-bhat** [comment]
@jackylk , @ravipesala : please check if this much info is enough or anything else need to be added.

File: docs/scd-and-cdc-guide.md (hunk 2 above)
Comment (ajantha-bhat) [reply docs/scd-and-cdc-guide.md:35]: Need to mention `insertHistoryTableExpr` action also ?

File: docs/scd-and-cdc-guide.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+   These clauses have the following semantics.
+    * `whenMatched` clauses can have at most one updateExpr and one delete action. The `updateExpr` action in merge only updates the specified columns of the matched target row. The `delete` action deletes the matched row.
+    * If there are two `whenMatched` clauses, then they are evaluated in order they are specified. The first clause must have a clause condition (otherwise the second clause is never executed).
+    * If both `whenMatched` clauses have conditions and neither of the conditions are true for a matching source-target row pair, then the matched target row is left unchanged.
+* `whenNotMatched` clause is executed when a source row does not match any target row based on the match condition.
+   * `whenNotMatched` clause can have only the `insertExpr` action. The new row is generated based on the specified column and corresponding expressions. Users do not need to specify all the columns in the target table. For unspecified target columns, NULL is inserted.
+* `whenNotMatchedAndExistsOnlyOnTarget` clause is executed when row does not match source and exists only in target. This clause can have only delete action.
+
```
Comment (ajantha-bhat) [inline docs/scd-and-cdc-guide.md:57]: what about other actions like `insertHistoryTableExpr` ?

Comment (Indhumathi27) [reply docs/scd-and-cdc-guide.md:57]: its implementation not found in carbon

File: docs/scd-and-cdc-guide.md (hunk 2 above)
Comment (Indhumathi27) [reply docs/scd-and-cdc-guide.md:35]: its implementation not found in carbon

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1258/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2976/

File: docs/scd-and-cdc-guide.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+than changing regularly on a time basis.
+
+SCD and CDC data changes can be merged to a carbon dataset online using the data frame level `MERGE` API.
+
+#### MERGE API
+
+Below API merges the datasets online and applies the actions as per the conditions. 
+
+```
+  targetDS.merge(sourceDS, <condition>)
```
Comment (ravipesala) [inline docs/scd-and-cdc-guide.md:35]: Unless added SQL syntax it won't be so understandable to users. The only way to understand is through example.

File: docs/scd-and-cdc-guide.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    * If both `whenMatched` clauses have conditions and neither of the conditions are true for a matching source-target row pair, then the matched target row is left unchanged.
+* `whenNotMatched` clause is executed when a source row does not match any target row based on the match condition.
+   * `whenNotMatched` clause can have only the `insertExpr` action. The new row is generated based on the specified column and corresponding expressions. Users do not need to specify all the columns in the target table. For unspecified target columns, NULL is inserted.
+* `whenNotMatchedAndExistsOnlyOnTarget` clause is executed when row does not match source and exists only in target. This clause can have only delete action.
+
+**NOTE:** SQL syntax for merge is not yet supported.
+
+##### Example code to implement cdc/scd scenario
+
+Please refer example class [MergeTestCase](https://github.com/apache/carbondata/blob/master/integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala) to understand and implement scd and cdc scenarios.
```
Comment (ravipesala) [inline docs/scd-and-cdc-guide.md:62]: I think we should create an example in the example module with the same test case content.

Comment (ravipesala) [reply docs/scd-and-cdc-guide.md:62]: Can create a jira and add in next releases.

Comment (Indhumathi27) [reply docs/scd-and-cdc-guide.md:62]: Created JIRA `CARBONDATA-3813` to add in next release.

**Comment by ajantha-bhat** [comment]
LGTM

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**